WebCryptoAPI/sign_verify/mldsa.js

```diff
@@ -0,0 +1,127 @@
+
+      // Check for successful verification even if data is altered after call.
+      promise_test(async() => {
+          let isVerified = false;
+          let key;
+          try {
+              key = await subtle.importKey("spki", vector.publicKeyBuffer, {name: vector.algorithmName}, false, ["verify"]);
+              var data = copyBuffer(vector.data);
+              [isVerified] = await Promise.all([
+                  subtle.verify(algorithm, key, vector.signature, data),
+                  data[0] = 255 - data[0]
+              ]);
+          } catch (err) {
+              assert_unreached("Verification should not throw error " + vector.name + ": " + err.message + "'");
+          }
+          assert_true(isVerified, "Signature verified");
+      }, vector.name + " with altered data after call");
+
+      // Check for successful signing and verification.
+      promise_test(async() => {
+          let signature;
+          let isVerified = false;
+          let privateKey, publicKey;
+          try {
+              privateKey = await subtle.importKey("pkcs8", vector.privateKeyBuffer, {name: vector.algorithmName}, false, ["sign"]);
+              publicKey = await subtle.importKey("spki", vector.publicKeyBuffer, {name: vector.algorithmName}, false, ["verify"]);
+
+              signature = await subtle.sign(algorithm, privateKey, vector.data);
+              isVerified = await subtle.verify(algorithm, publicKey, signature, vector.data);
+          } catch (err) {
+              assert_false(privateKey === undefined, "importKey failed for private key in " + vector.name + ". Message: ''" + err.message + "''");
+              assert_false(publicKey === undefined, "importKey failed for public key in " + vector.name + ". Message: ''" + err.message + "''");
+              assert_unreached("Sign/verify should not throw error " + vector.name + ": " + err.message + "'");
+          }
+          assert_not_equals(signature, undefined, "Signature should be produced");
+          assert_true(isVerified, "Round trip signature/verification should work");
+      }, vector.name + " round trip");
+
+      // Check for successful signing and verification with altered signature after sign.
+      promise_test(async() => {
+          let signature;
+          let isVerified = false;
+          let privateKey, publicKey;
+          try {
+              privateKey = await subtle.importKey("pkcs8", vector.privateKeyBuffer, {name: vector.algorithmName}, false, ["sign"]);
+              publicKey = await subtle.importKey("spki", vector.publicKeyBuffer, {name: vector.algorithmName}, false, ["verify"]);
+
+              signature = await subtle.sign(algorithm, privateKey, vector.data);
+              var originalSignature = copyBuffer(signature);
+              signature[0] = 255 - signature[0]; // Alter signature
+              isVerified = await subtle.verify(algorithm, publicKey, originalSignature, vector.data);
```

Comment:
I don't think this one tests what it's intended to; should probably be something like
```suggestion
              const promise = subtle.verify(algorithm, publicKey, signature, vector.data);
              signature[0] = 255 - signature[0]; // Alter signature
              isVerified = await promise;
```
or as you did above:
```suggestion
              [isVerified] = await Promise.all([
                  subtle.verify(algorithm, publicKey, signature, vector.data),
                  signature[0] = 255 - signature[0]
              ]);
```